### PR TITLE
HDDS-5543: OMKeyRequest#createFileInfo should handle ECReplicationConfig

### PR DESCRIPTION
**Pull request by umamaheswararao**
## What changes were proposed in this pull request?

When createKey, we have to handle EC replication config. This JIRA addressing that issue.

## What is the link to the Apache JIRA

https://issues.apache.org/jira/browse/HDDS-5543

## How was this patch tested?

Added a test case verify the behavior.

### CODE REVIEW

**Comment by umamaheswararao** [comment]
Please note there is a small code overlap from HDDS-5510. Once HDDS-5510 committed I will remove that part (in ContainerManagerImpl).

File: hadoop-ozone/integration-test/src/test/java/org/apache/hadoop/ozone/client/rpc/TestECKeyOutputStream.java [hunk 1]
```diff
@@ -0,0 +1,113 @@
...
+   */
+  @AfterClass
+  public static void shutdown() {
+    if (cluster != null) {
+      cluster.shutdown();
+    }
+  }
+
+  @Test
+  public void testCreateKeyWithECReplicationConfig() throws Exception {
```
Comment (sodonnel) [inline hadoop-ozone/integration-test/src/test/java/org/apache/hadoop/ozone/client/rpc/TestECKeyOutputStream.java:106]: I spoke to Uma about this test - at this stage it just obtains the output stream and hence the container is allocated through OM to SCM. However no data is written as the there is still an issue with HDDS-5425 which will prevent the write completing. For me, this is OK at this stage, as we are still actively developing this and we are on branch.

**Review by sodonnel — APPROVED** [review]
+1 pending green CI.

File: hadoop-ozone/integration-test/src/test/java/org/apache/hadoop/ozone/client/rpc/TestECKeyOutputStream.java (hunk 1 above)
Comment (umamaheswararao) [reply hadoop-ozone/integration-test/src/test/java/org/apache/hadoop/ozone/client/rpc/TestECKeyOutputStream.java:106]: Thanks @sodonnel for the review. This new test class will used eventually for full write after HDDS-5425 done.

**Pull request merged**